src/scribe_data/translation/translation_utils.py

```diff
@@ -24,14 +24,77 @@
 import os
 import signal
 import sys
+from functools import lru_cache
+from typing import List
+from pathlib import Path
 
 from transformers import M2M100ForConditionalGeneration, M2M100Tokenizer
+from scribe_data.wikidata.wikidata_utils import JSON, sparql
 
 from scribe_data.utils import (
     get_language_iso,
     get_target_langcodes,
 )
 
+LANGUAGE_METADATA_FILE = (
+    Path(__file__).parent.parent / "resources" / "language_metadata.json"
+)
+
+with LANGUAGE_METADATA_FILE.open("r", encoding="utf-8") as file:
+    language_data = json.load(file)
+
+
+@lru_cache(maxsize=None)
+def get_language_qid(language_name: str) -> str:
+    """Get the QID for a given language name."""
+    normalized_name = language_name.lower()
+    for language in language_data["languages"]:
+        if language["language"] == normalized_name:
+            return language["qid"]
+    return None
+
+
+@lru_cache(maxsize=None)
+def get_all_articles(language: str) -> List[str]:
+    """Get all articles for a given language."""
+    qid_from_language_metadata = get_language_qid(language)
+    query = f"""
+    SELECT DISTINCT ?article WHERE {{
+      VALUES ?language {{ wd:{qid_from_language_metadata} }}
+      ?lexeme dct:language ?language ;
+              wikibase:lexicalCategory ?category ;
+              wikibase:lemma ?lemma .
+      VALUES ?category {{ wd:Q2865743 wd:Q3813849 wd:Q576670}}  # Definite, indefinite articles and Partitive Articles
+
+      {{
+        ?lexeme wikibase:lemma ?article .
+      }} UNION {{
+        ?lexeme ontolex:lexicalForm ?form .
+        ?form ontolex:representation ?article .
+      }}
+    }}
+    """
+
+    sparql.setQuery(query)
+    sparql.setReturnFormat(JSON)
+    results = sparql.query().convert()
+    return [result["article"]["value"] for result in results["results"]["bindings"]]
+
+
+def remove_articles_from_words(
+    batch_words: List[str], articles: List[str]
+) -> List[str]:
+    """Remove articles from a given list of words."""
+
+    def remove_article(word: str) -> str:
+        for article in articles:
+            if word.lower().startswith(article.lower() + " "):
+                print(f"Article '{article}' found in word: '{word}'")
```

Comment:
We don't need the print statement here for each of the instances of articles being found :)

Reply:
I have this removed locally 😊